File: lab01/builder/src/main/java/com/example/builder/director/CharacterDirector.java
package com.example.builder.director;

import com.example.builder.builder.CharacterBuilder;
import com.example.builder.dto.CharacterBuildRequest;
import com.example.builder.model.Character;

public class CharacterDirector {
    public Character construct(CharacterBuilder builder, CharacterBuildRequest request) {

        return builder.reset()
                .setName(request.getName())
                .setHealth(request.getHealth())
                .setMana(request.getMana())
                .setWeapon(request.getWeapon())
                .build();
    }
}

File: lab01/builder1/Director/CharacterDirector.cs
using Builder1.Builders;
using Builder1.Dto;
using Builder1.Models;

namespace Builder1.Director;

public class CharacterDirector
{
    private ICharacterBuilder? _builder;

    public void SetBuilder(ICharacterBuilder builder) => _builder = builder;

    public Character Make(CharacterBuildRequest request)
    {
        if (_builder is null)
            throw new InvalidOperationException("Builder is not set");

        return _builder.Reset()
            .SetName(request.Name)
            .SetHealth(request.Health)
            .SetMana(request.Mana)
            .SetWeapon(request.Weapon)
            .Build();
    }
}
